packages/botonic-react/src/utils.js

```diff
@@ -8,6 +11,47 @@ import { Video } from './components/video'
 import { Document } from './components/document'
 import { Location } from './components/location'
 
+export function decomposeComponent(component) {
+  let componentJSON = null
+  let carousel = []
+  try {
+    component[0].props.children.map((e, i) => {
+      let c = e.props.children
+      carousel[i] = {
+        img: c[0].props.src,
+        title: c[1].props.children,
+        subtitle: c[2].props.children,
+        button: c[3].props
+      }
+    })
+  } catch (e) {}
+  if (carousel.length) {
+    componentJSON = carousel
+  }
+  try {
```

Comment:
Maybe we can have a single try/catch for all cases?
Would a switch be more clear?

Reply:
Sure, I am going to refactor the code with a switch and we will only need a single try/catch. Good point!

Reply:
👍